corehq/apps/tour/models.py

```diff
@@ -0,0 +1,41 @@
+import collections
+from django.core import cache
+
+
+GuidedTourStep = collections.namedtuple('GuidedTourStep', ['element', 'title', 'content'])
+
+
+class StaticGuidedTour(object):
+
+    def __init__(self, slug, steps):
+        self.slug = slug
+        if not isinstance(steps, list):
+            raise ValueError("steps should be a list of GuidedTourStep")
+        self.steps = steps
+
+    def get_tour_data(self):
+        return {
+            'steps': map(lambda s: dict(s._asdict()), self.steps),
+        }
+
+    def _get_cache_key(self, request, domain):
+        return "{}.{}.{}".format(domain, request.couch_user.username, self.slug)
```

Comment:
minor - if this only needs the username it might make sense to make that the argument since passing the whole request in makes the footprint of the function much larger (if you wanted the convenience of only calling `request.couch_user.username` in one place you could make a pass-through function that takes the request and calls the second one)